document-store/src/main/java/org/hypertrace/core/documentstore/postgres/query/v1/vistors/PostgresFilterTypeExpressionVisitor.java

```diff
@@ -44,16 +45,15 @@ public String visit(final RelationalExpression expression) {
     RelationalOperator operator = expression.getOperator();
     SelectTypeExpression rhs = expression.getRhs();
 
-    // Only an identifier LHS and a constant RHS is supported as of now.
+    // Only a constant RHS is supported as of now.
     PostgresSelectTypeExpressionVisitor rhsVisitor = new PostgresConstantExpressionVisitor();
     Object value = rhs.accept(rhsVisitor);
-    PostgresIdentifierExpressionVisitor identifierVisitor =
-        new PostgresIdentifierExpressionVisitor();
-    PostgresSelectTypeExpressionVisitor lhsVisitor =
-        new PostgresDataAccessorIdentifierExpressionVisitor(postgresQueryParser, getType(value));
 
-    final String fieldName = lhs.accept(identifierVisitor);
-    final String parsedLhsExpression = lhs.accept(lhsVisitor);
+    PostgresRelationalFilterLhsVisitor lhsVisitor =
+        new PostgresRelationalFilterLhsVisitor(postgresQueryParser, getType(value));
+
+    final RelationalFilterLhsParseResult parseResult = lhs.accept(lhsVisitor);
```

Comment:
The `RelationalFilterLhsParseResult` is providing `fieldName` and `parsedExpression` same value with parsed.
e.g
`fieldName = CAST (document->>'quantity' AS NUMERIC) * CAST (document->>'price' AS NUMERIC)`
`ParsedExpression = CAST (document->>'quantity' AS NUMERIC) * CAST (document->>'price' AS NUMERIC)`

So, we have to use `prepareParsedNonCompositeFilter` from PostgresUtils otherwise few operators like `NEQ` will not work as expected.
.eg 
`CAST (document->>'quantity' AS NUMERIC) * CAST (document->>'price' AS NUMERIC) != 10`

Reply:
So, would suggest not setting `fieldName` in the result set of `RelationalFilterLhsParseResult`. And doing as below:

```
if (fieldName != null) {
      FieldToPgColumn fieldToPgColumn =
          postgresQueryParser.getToPgColumnTransformer().transform(fieldName);

      if (fieldToPgColumn.getTransformedField() == null)
        throw new UnsupportedOperationException("jsonb types in where clause is not yet supported");

      return PostgresUtils.parseNonCompositeFilter(
          fieldToPgColumn.getTransformedField(),
          parseResult.getParsedExpression(),
          fieldToPgColumn.getPgColumn(),
          operator.toString(),
          value,
          postgresQueryParser.getParamsBuilder());
    } else {
      return PostgresUtils.prepareParsedNonCompositeFilter(
          parseResult.getParsedExpression(),
          operator.toString(),
          value,
          postgresQueryParser.getParamsBuilder());
    }
```

Reply:
Did some simplification of code. Please check if that works. Otherwise, I'll revert it based on the null check.